eng/native/configurecompiler.cmake

```
@@ -693,9 +693,15 @@ if (CLR_CMAKE_HOST_UNIX)
     endif()
   endif(CLR_CMAKE_HOST_MACCATALYST)
 
+  if(CLR_CMAKE_TARGET_ARCH_ARM64 AND CLR_CMAKE_TARGET_LINUX)
```

```suggestion
  if(NOT CLR_CROSS_COMPONENTS_BUILD AND CLR_CMAKE_TARGET_ARCH_ARM64 AND CLR_CMAKE_TARGET_LINUX)
```

I think we should exclude this case. The cross components are built for the host machine (x64), even during cross-compilation (host: x64, target: arm64). After building libcoreclr.so for ARM64, it fails in the secondary tools build because the -mbranch-protection flag does not exist.

---

Ooh, interesting! Thanks for the fix 🙂 .